crates/matrix-sdk/src/room/power_levels.rs

```diff
@@ -0,0 +1,289 @@
+//! Power level configuration types used in [the `room` module][super].
+
+use ruma::events::{room::power_levels::RoomPowerLevels, StateEventType};
+
+/// The power level settings required for various operations within a room.
+/// When updating these settings, any levels that are `None` will remain
+/// unchanged.
+#[derive(Debug)]
+#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
+pub struct RoomPowerLevelSettings {
+    // Actions
+    /// The level required to ban a user.
+    pub ban: Option<i64>,
+    /// The level required to invite a user.
+    pub invite: Option<i64>,
+    /// The level required to kick a user.
+    pub kick: Option<i64>,
+    /// The level required to redact an event.
+    pub redact: Option<i64>,
+
+    // Events
+    /// The default level required to send message events.
+    pub events_default: Option<i64>,
+    /// The default level required to send state events.
+    pub state_default: Option<i64>,
+    /// The default power level for every user in the room.
+    pub users_default: Option<i64>,
+    /// The level required to change the room's name.
+    pub room_name: Option<i64>,
+    /// The level required to change the room's avatar.
+    pub room_avatar: Option<i64>,
+    /// The level required to change the room's topic.
+    pub room_topic: Option<i64>,
+}
+
+impl From<RoomPowerLevels> for RoomPowerLevelSettings {
+    fn from(value: RoomPowerLevels) -> Self {
+        Self {
+            ban: Some(value.ban.into()),
+            invite: Some(value.invite.into()),
+            kick: Some(value.kick.into()),
+            redact: Some(value.redact.into()),
+            events_default: Some(value.events_default.into()),
+            state_default: Some(value.state_default.into()),
+            users_default: Some(value.users_default.into()),
+            room_name: value
+                .events
+                .get(&StateEventType::RoomName.into())
+                .map(|v| (*v).into())
+                .or(Some(value.state_default.into())),
```

Comment:
Equally, I'm split on whether this nil coalescing should happen client-side or in the SDK.